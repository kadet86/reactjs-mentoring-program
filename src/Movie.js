import React from 'react';

export default function Movie({movie}) {
    return (
        <article className="movie">
            <img src={movie.poster_path} width="200" />
            <div className="movie__description">
                <h2>{movie.title}</h2>
                <div>{movie.vote_average}</div>
                <div>{movie.tagline}</div>
                <div>{movie.release_date.substr(0, 4)}</div>
                <div>{movie.runtime} min</div>
                <div>{movie.overview}</div>
            </div>
        </article>
    );
}