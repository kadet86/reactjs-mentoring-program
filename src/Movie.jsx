// @flow
import * as React from 'react';

type MovieInfo = {
    poster_path: string,
    title: string,
    vote_average: string,
    tagline: string,
    release_date?: string,
    runtime?: number,
    overview: string,
};
type Props = {
    movie: MovieInfo,
};

export default function Movie({ movie }: Props) {
    return (
        <article className="movie">
            <img src={movie.poster_path} />
            <div className="movie__description">
                <h2>{movie.title}</h2>&nbsp;
                <div className="movie__rating">{movie.vote_average}</div>
                <div className="movie__tagline">{movie.tagline}</div>
                <span className="movie__date">
                    {movie.release_date && movie.release_date.substr(0, 4)}
                </span>
                <span className="movie__duration">
                    {movie.runtime || 0} min
                </span>
                <div className="movie__overview">{movie.overview}</div>
            </div>
        </article>
    );
}
