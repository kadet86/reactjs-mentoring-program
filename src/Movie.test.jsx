import React from 'react';
import {shallow} from 'enzyme';
import Movie from './Movie';

describe('Movie component', () => {
    it('has persistent snapshot with movie={{}}', () => {
        const component = shallow(<Movie movie={{}}/>);
        expect(component.html()).toMatchSnapshot();
    });

    it('has persistent snapshot with movie={{release_date:"2018-02-03"}}', () => {
        const component = shallow(<Movie movie={{release_date:"2018-02-03"}}/>);
        expect(component.html()).toMatchSnapshot();
    });
});
