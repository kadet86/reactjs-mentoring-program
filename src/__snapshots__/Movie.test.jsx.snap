// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component has persistent snapshot with movie={{}} 1`] = `ShallowWrapper {}`;

exports[`Movie component has persistent snapshot with movie={{release_date:"2018-02-03"}} 1`] = `ShallowWrapper {}`;
